app/models/patient.rb

```diff
@@ -744,49 +744,53 @@ def self.order_by_risk(asc: true)
   # e.g. Jon Smith M null would be a duplicate of Jon Smith M 1/1/2000
   # OR
   # - matching state/local id
-  def self.duplicate_data(first_name, last_name, sex, date_of_birth, user_defined_id_statelocal)
+  def self.duplicate_data_detection(patient)
     # Track which matches have occurred
     duplicate_field_data = []
 
     # check for a duplicate state/local id
-    dup_statelocal_id = where('user_defined_id_statelocal = ?', user_defined_id_statelocal&.to_s&.strip)
+    if patient[:user_defined_id_statelocal].present?
+      dup_statelocal_id = where('trim(user_defined_id_statelocal) = ?', patient[:user_defined_id_statelocal]&.to_s&.strip)
+    end
     duplicate_field_data << { count: dup_statelocal_id.size, fields: ['State/Local ID'] } if dup_statelocal_id.present?
 
     # if first_name or last_name is null skip duplicate detection
-    return { is_duplicate: duplicate_field_data.any?, duplicate_field_data: duplicate_field_data } if first_name.nil? || last_name.nil?
+    return { is_duplicate: duplicate_field_data.any?, duplicate_field_data: duplicate_field_data } if patient[:first_name].nil? || patient[:last_name].nil?
 
     fn_ln_sex_dob_matches = 0
     fn_ln_sex_matches = 0
     fn_ln_dob_matches = 0
     fn_ln_matches = 0
 
+    # The EPI-X format sends this as a Date while Exposure sends it as a string
+    date_of_birth = patient[:date_of_birth].instance_of?(Date) ? patient[:date_of_birth]&.strftime('%F') : patient[:date_of_birth]
+
     # Determine which type of duplicate exists
-    where(first_name: first_name, last_name: last_name).pluck(*%i[sex date_of_birth]).each do |(s, dob)|
+    where(first_name: patient[:first_name], last_name: patient[:last_name])
+      .select('sex', "DATE_FORMAT(date_of_birth, '%Y-%m-%d') date_of_birth").pluck(:sex, :date_of_birth).each do |(s, dob)|
       dob = dob&.strftime('%F')
```

Comment:
Since you added the SQL format you don't need this line anymore.

Reply:
Are you referring to `dob = dob&.strftime('%F')`? I think I need it because it comes back as a date when the date_of_birth is sometimes a date and other times a string; I converted date_of_birth to a string above the loop. Is there a better way to do this?